tests/scripts/all.sh

```diff
@@ -590,7 +590,7 @@ pre_setup_keep_going () {
         case "$1" in
             "msg "*) false;;
             "cd "*) false;;
-            *make*[\ /]tests*) false;; # make tests, make CFLAGS=-I../tests, ...
+            *make*[\ /]tests*) false;; # make tests, make CFLAGS=-Itests, ...
```

Comment:
I'm afraid that if we need to make this change in our scripts, it likely means some users will have to make similar changes, and so this breaks backward compatibility.

Reply:
In principle, I _like_ that everything is done from the toplevel directory. I'm just worried that we can't do it.

I think we can preserve backward compatibility at the cost of adding a lot of `cd` commands.

Reply:
Hmmm, Ok. I think I know how I could emulate this behaviour, If we want to keep maximum backwards compatibility.

Reply:
On second thought, the current behaviour of our Makefiles with regards to `CFLAGS`, etc. is very unintuitive, if you use relative paths.

`CFLAGS=-Ifoo` means `library/foo` for the files in the `library` and `3rdparty/everest` directories, while it means `programs/foo` in the `programs` directory, `tests/foo` in the tests directory and `tests/fuzz/foo` in the `tests/fuzz/foo` directory.

It would be much more sane to be relative to the toplevel directory. I would expect that users have to work around this already by using absolute paths if they build anything outside the files under `library`.

Reply:
Granted, most users probably *do* only care about things under `library`

Reply:
I've added the cd commands needed to emulate the old behaviour.